tracer/src/Datadog.Trace/Util/Http/QueryStringObfuscation/Obfuscator.cs

```diff
@@ -16,11 +24,27 @@ internal class Obfuscator : ObfuscatorBase
         private readonly TimeSpan _timeout;
         private readonly IDatadogLogger _logger;
 
-        internal Obfuscator(string pattern, TimeSpan timeout, IDatadogLogger logger)
+        internal Obfuscator(string pattern, TimeSpan timeout, IDatadogLogger logger, bool supportBacktracking)
         {
+#if NETCOREAPP3_1_OR_GREATER
+            AppDomain.CurrentDomain.SetData("REGEX_NONBACKTRACKING_MAX_AUTOMATA_SIZE", 2000);
+#endif
             _timeout = timeout;
             _logger = logger;
-            _regex = new(pattern, RegexOptions.IgnoreCase | RegexOptions.Compiled, _timeout);
+
+            var options =
+                    RegexOptions.IgnoreCase
+                    | RegexOptions.Compiled;
+
+#if NETCOREAPP3_1_OR_GREATER
+        if (!supportBacktracking)
+        {
+            options |= RegexOptions.NonBacktracking
+        }
+#endif
```

Comment:
All these negatives are a bit confusing, given the _feature_ is non-backtracking. I think `supportBacktracking` should be `supportsNonBacktracking` instead, which defaults to false

Reply:
I agree, possibly the best option is to remove this switch, as it's only used for testing a timeout. See comment below.

Reply:
In the end, did as you suggested in 3caf7d25f

Reply:
@robertpi Given you inverted the variable, shouldn't the default value in `ObfuscatorFactory` have been inverted too?